### PR TITLE
XIVY-2250 Dynamic Dialog Demo

### PR DESCRIPTION
**Pull request by ivy-tgr**

### CODE REVIEW

File: html-dialog/html-dialog-demos/webContent/layouts/frame-10.xhtml
Comment (ivy-rew) [inline html-dialog/html-dialog-demos/webContent/layouts/frame-10.xhtml:1]: omit this file; use the template all other HDs are using instead. 
it lives in the /webContent

File: html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+<h:body>
+ <ui:composition template="/layouts/HtmlDemo.xhtml">
+    <ui:param name="centerHeader" value="Dynamic Dialog Demo" />
+    <ui:define name="breadcrumb">Dynamic Dialog Demo</ui:define>
+    <ui:define name="content">
+    
+    <h:panelGroup layout="block">
+        <p>Dynamic dialogs allows you to open xhtml files in new dialogs.
+           This helps you to maintain the dialogs.
+           It's easier to test and the view state is smaller and saves memory.</p>
```
Comment (ivy-rew) [inline html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml:17]: I'd prefer a bullet-point list with the benefits.

File: html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    xmlns:p="http://primefaces.org/ui"
+    xmlns:pe="http://primefaces.org/ui/extensions">
+<h:body>
+ <ui:composition template="/layouts/HtmlDemo.xhtml">
+    <ui:param name="centerHeader" value="Dynamic Dialog Demo" />
+    <ui:define name="breadcrumb">Dynamic Dialog Demo</ui:define>
+    <ui:define name="content">
+    
+    <h:panelGroup layout="block">
+        <p>Dynamic dialogs allows you to open xhtml files in new dialogs.
```
Comment (ivy-rew) [inline html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml:15]: please use a spell-checker to correct the grammer.
e.g. https://app.grammarly.com/docs/new

File: html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemoData.ivyClass [hunk 3]
```diff
@@ -0,0 +1,4 @@
+DynamicDialogDemoData #class
+ch.ivyteam.htmldialog.demo.output.DynamicDialogDemo #namespace
+data String #field
```
Comment (ivy-rew) [inline html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemoData.ivyClass:3]: ?

**Review by ivy-rew — COMMENTED** [review]
good direction; please add a webtest to ensure that your demo actually works.

File: html-dialog/html-dialog-demos/webContent/layouts/frame-10.xhtml
Comment (ivy-tgr) [reply html-dialog/html-dialog-demos/webContent/layouts/frame-10.xhtml:1]: They all have the sidebar & a header in the template, so I can't use them for the dialog.
grafik
grafik

File: html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    xmlns:pe="http://primefaces.org/ui/extensions">
+<h:body>
+  <ui:composition template="/layouts/HtmlDemo.xhtml">
+    <ui:param name="centerHeader" value="Dynamic Dialog Demo" />
+    <ui:define name="breadcrumb">Dynamic Dialog Demo</ui:define>
+    <ui:define name="content">
+
+      <h:panelGroup layout="block">
+
+        <p>Dynamic dialogs allows you to open xhtml files in new dialogs.</p>
```
Comment (ivy-rew) [inline html-dialog/html-dialog-demos/src_hd/ch/ivyteam/htmldialog/demo/output/DynamicDialogDemo/DynamicDialogDemo.xhtml:16]: grammarly

**Pull request merged**